cocos/2d/CCLabel.cpp

```diff
@@ -1287,45 +1393,63 @@ void Label::setSystemFontSize(float fontSize)
 ///// PROTOCOL STUFF
 Sprite * Label::getLetter(int letterIndex)
 {
-    if (_systemFontDirty || _currentLabelType == LabelType::STRING_TEXTURE)
+    Sprite* letter = nullptr;
+    do 
     {
-        return nullptr;
-    }
+        if (_systemFontDirty || _currentLabelType == LabelType::STRING_TEXTURE)
+        {
+            break;
+        }  
 
-    if (_contentDirty)
-    {
-        updateContent();
-    }
-    
-    if (! _textSprite && letterIndex < _limitShowCount)
-    {
-        const auto &letter = _lettersInfo[letterIndex];
+        auto contentDirty = _contentDirty;
+        if (contentDirty)
+        {
+            updateContent();
+        }
 
-        if(! letter.def.validDefinition)
-            return nullptr;
+        if (_textSprite == nullptr && letterIndex < _limitShowCount)
+        {
+            const auto &letterInfo = _lettersInfo[letterIndex];
+            if (!letterInfo.def.validDefinition)
+            {
+                break;
+            } 
 
-        Sprite* sp = static_cast<Sprite*>(this->getChildByTag(letterIndex));
+            if (_letters.find(letterIndex) != _letters.end())
```

Comment:
The function `getLetter` is very complex, please retrieve some logic and make it more clearer.
Thanks.